proto/buf/alpha/registry/v1alpha1/repository_branch.proto

```diff
@@ -39,7 +39,13 @@ service RepositoryBranchService {
   rpc ListRepositoryBranches(ListRepositoryBranchesRequest) returns (ListRepositoryBranchesResponse) {
     option idempotency_level = NO_SIDE_EFFECTS;
   }
-  // GetCurrentDefaultBranch returns the branch name that is mapped to the main/BSR_HEAD.
+  // GetCurrentDefaultBranch returns the branch name that is mapped to the main/BSR_HEAD. This might
+  // not be the same value in the repository's `default_branch` field, since that value can be
+  // changed at will by repository's owners/admins for syncing git repositories. This RPC retrieves
+  // the branch from the latest commit labeled as BSR_HEAD, even if that value differs from the one
+  // stored in the `default_branch` field.
+  //
+  // TODO: Rename this RPC to something more appropriate like "GetLatestHEADCommit".
```

Comment:
We can probably remove or deprecate it instead, but we can decide later outside of this pr.

Reply:
Agree, I'll leave it to @doriable and you to coordinate on that one 😄